### PR TITLE
Remove end2end.yml

### PR DESCRIPTION
**Pull request by pingsutw**
## Tracking issue
NA

## What changes were proposed in this pull request?
We already have end2end tests [here](https://github.com/flyteorg/flyte/blob/faf8fc8153b3e0ac039c300ceab31c54be537985/.github/workflows/single-binary.yml#L200), so we don't need end2end.yml anymore

## Check all the applicable boxes 

- [ ] I updated the documentation accordingly.
- [ ] All new and existing tests passed.
- [ ] All commits are signed-off.

## Related PRs
https://github.com/flyteorg/flytectl/pull/462

### CODE REVIEW

**Comment by eapolinario** [comment]
I'm 99% sure we don't use this workflow anywhere. Can you double-check and if that's the case, should we remove it?

**Comment by pingsutw** [comment]
@eapolinario I revive end2end test. It only runs when merging PRs since it takes 15 minutes.

https://github.com/flyteorg/flyte/actions/runs/8967710966/job/24626449666?pr=5034

File: boilerplate/flyte/end2end/run-tests.py [hunk 1]
```diff
@@ -243,6 +248,12 @@ def run(
     is_flag=False,
     help="Name of domain in project to run functional tests on",
 )
+@click.option(
+    "--parallel",
+    is_flag=True,
+    default=False,
```
Comment (pingsutw) [inline boilerplate/flyte/end2end/run-tests.py:254]: Run flytesnacks workflow serially in the CI due to disk pressure in the sandbox.

**Review by eapolinario — APPROVED** [review]
That's pretty cool. Thank you!

File: .github/workflows/end2end.yml [hunk 2]
```diff
@@ -30,40 +27,10 @@ jobs:
...
-          cp /tmp/tmp/docker-images-flytepropeller/snapshot-flytepropeller.tar snapshot-flytepropeller.tar
           flytectl config init
-          flytectl sandbox start --source=$(pwd)
-      - name: Prime docker cache
-        run: |
-          flytectl sandbox exec -- docker load -i /root/snapshot-datacatalog.tar
-          flytectl sandbox exec -- docker load -i /root/snapshot-flyteadmin.tar
-          flytectl sandbox exec -- docker load -i /root/snapshot-flytecopilot.tar
-          flytectl sandbox exec -- docker load -i /root/snapshot-flytepropeller.tar
+          flytectl demo start --source=$(pwd) --image ghcr.io/${{ github.repository_owner }}/flyte-sandbox-bundled:latest --disable-agent
```
Comment (eapolinario) [inline .github/workflows/end2end.yml:33]: just to confirm, where is this image built?

**Review by eapolinario — CHANGES_REQUESTED** [review]

File: .github/workflows/single-binary.yml [hunk 3]
```diff
@@ -246,3 +246,11 @@ jobs:
           tags: ${{ steps.image-names.outputs.tags }}
           build-args: "FLYTE_SANDBOX_VERSION=${{ env.FLYTE_SANDBOX_VERSION }}"
           push: true
+
+  endtoend:
+    name: End2End Test
+    needs: [ build-and-push-sandbox-bundled-image ]
```
Comment (pingsutw) [inline .github/workflows/single-binary.yml:252]: @eapolinario here. after `build-and-push-sandbox-bundled-image`

**Review by eapolinario — DISMISSED** [review]

**Review by eapolinario — APPROVED** [review]

**Comment by pingsutw** [comment]
Just found that we already have end2end tests [here](https://github.com/flyteorg/flyte/blob/faf8fc8153b3e0ac039c300ceab31c54be537985/.github/workflows/single-binary.yml#L200), so we don't need `end2end.yml` anymore

**Review by Future-Outlier — APPROVED** [review]

**Review by eapolinario — APPROVED** [review]

**Pull request merged**